central/complianceoperator/v2/remediations/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,112 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	remediationStorage "github.com/stackrox/rox/central/complianceoperator/v2/remediations/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/sac/testutils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+func TestComplianceRemediationDataStore(t *testing.T) {
+	suite.Run(t, new(complianceRemediationDataStoreTestSuite))
+}
+
+type complianceRemediationDataStoreTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+
+	hasReadCtx            context.Context
+	hasWriteCtx           context.Context
+	noAccessCtx           context.Context
+	testContexts          map[string]context.Context
+	nonComplianceContexts map[string]context.Context
+
+	dataStore DataStore
+	storage   remediationStorage.Store
+	db        *pgtest.TestPostgres
+}
+
+func (s *complianceRemediationDataStoreTestSuite) SetupSuite() {
+	s.T().Setenv(features.ComplianceEnhancements.EnvVar(), "true")
+	s.T().Setenv(features.ComplianceRemediationV2.EnvVar(), "true")
+	if !features.ComplianceEnhancements.Enabled() {
+		s.T().Skip("Skip tests when ComplianceEnhancements disabled")
+		s.T().SkipNow()
+	}
+}
+
+func (s *complianceRemediationDataStoreTestSuite) SetupTest() {
+	s.hasReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Compliance)))
+	s.hasWriteCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Compliance)))
+	s.noAccessCtx = sac.WithGlobalAccessScopeChecker(context.Background(), sac.DenyAllAccessScopeChecker())
+	s.testContexts = testutils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Compliance)
+	s.nonComplianceContexts = testutils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Deployment)
```

Comment:
Are these used? I think they can be deleted.

Reply:
done